pkg/gardenlet/apis/config/v1alpha1/types.go

```diff
@@ -347,7 +347,7 @@ type LeaderElectionConfiguration struct {
 
 // SeedConfig contains configuration for the seed cluster.
 type SeedConfig struct {
-	gardencorev1beta1.Seed `json:",inline"`
+	gardencorev1beta1.SeedTemplate `json:",inline"`
```

Comment:
WDYT about removing the `SeedConfig` type in favor of the new `SeedTemplate`?

We might even consider renaming the `seedConfig` field to `seedTemplate` to make it consistent with the `ManagedSeed`.
Yes, I know, this is a breaking change. But the gardenlet config API is only in an `alpha` version, which means we are able to introduce breaking changes by API conventions.
Though, this is probably too much effort for such a cosmetic change and is not worth it.
WDYT?

Reply:
Yes, I think it's not worth it. As it is an existing YAML gardenlet configuration is basically compatible, even if a different type is being used. To introduce this we should either create a new version (`v1alpha2`) and take care of conversions, or we should be prepared to break all existing gardenlet configurations ... I don't think we want to spend the time and effort to fix them all.

Reply:
Yes, agreed.

What about the first question: should we remove the `SeedConfig` type and change the `seedConfig` field to be of type `SeedTemplate` (no indirection)?

Reply:
I don't like `seedConfig` of type `SeedTemplate`, and I don't think the indirection hurts in any way. I would vote for no more changes.

Could we merge this PR already?

Reply:
Ok, fine for me.